bismark

```diff
@@ -3050,6 +3049,9 @@ sub check_bowtie_results_single_end_bowtie2{
 		  # warn "$first_ambig_alignment\n"; sleep(1);
 	      }
 	  }
+	  elsif ($alignment_score == $best_AS_so_far) {
+	      $amb_same_thread = 1;
+	  }
```

Comment:
I don't think this is the right way to go about this because if this alignment has the same alignment score as the best AS so far then this is not ambiguous in the same thread. To be ambiguous within the same thread the AS and the second best hit have to have the same score.
If overwrite is specified it will be added to a the alignment data structure further down, and in a later step all alignments in that data structure will be sorted by their lowest AS. If two different locations have the same lowest AS the read is booted altogether.